### ISSUE TITLE
[Question] 0.4.0 release on npm?

### ISSUE THREAD

**Issue by AltF02** [cdmoro/bootstrap-vue-3#771]
**What you are trying to do**
I am trying to update to the most recent version (0.4.0)

**What you have already tried**
I could provide the direct git link, but its missing on npmjs.org

**Comment by VividLemon** [comment]
Occasionally the ci breaks with this issue.
![image](https://user-images.githubusercontent.com/31498096/198644028-5a6ab782-0402-49b9-8cd5-9686c4308e08.png)

**Comment by AltF02** [comment]
It could be that because you are tracking `pnpm-lock.json` when [pnpm install --ignore-scripts](https://github.com/cdmoro/bootstrap-vue-3/blob/main/.github/workflows/npm-publish.yaml#L33) runs. That it gets updated with new packages not yet indexed in the lockfile on HEAD. 

A solution would be adding `--frozen-lockfile` to `pnpm install` as found [here](https://pnpm.io/cli/install)

**Comment by VividLemon** [comment]
![image](https://user-images.githubusercontent.com/31498096/198677788-09040b96-1989-445d-83d6-8cc07461c62f.png)

**Comment by AltF02** [comment]
:/ https://github.com/cdmoro/bootstrap-vue-3/actions/runs/3346668168/jobs/5543758456 seems like it failed anyways. Let me try if I can replicate it locally

**Comment by AltF02** [comment]
I created #773, not intended to be merged, but could give us some more answers

**Comment by VividLemon** [comment]
![image](https://user-images.githubusercontent.com/31498096/198684078-250a0642-cef2-47fc-b526-d08a0b3c923f.png)
wot

**Comment by AltF02** [comment]
hmm that's odd, I guess you could add `--no-git-checks` although its a hacky solution

**Comment by VividLemon** [comment]
The unknown branch issue is different because when a release is manually created, actions/checkout@v3 does not automatically check out the main branch ~~ it checks out HEAD. 

The git unclean issue is still an issue, but doesn't seem to be occurring at the moment. 

But the main issue right now is that npm is failing to authenticate when running publish. I think pnpm may have made a change recently that affected this.

**Comment by alexdohm** [comment]
so im noticing that this is still not available with 0.4.3...should it be accessible now over npm? sorry if youre still working on this!

**Comment by HenryUxDesugner** [comment]
any luck or can you give the steps to build and use locally?

**Comment by VividLemon** [comment]
The source is available on each release https://github.com/cdmoro/bootstrap-vue-3/releases/tag/v0.4.4 . You can download it, unpack it, open the new folder, run pnpm install, pnpm build, then cut out the ./packages/bootstrap-vue-3 folder. At that point, the folder is an exact copy of what you'd get in node_modules. How you would use this folder is unknown to me. It could go into node_modules, but idk

**Comment by AltF02** [comment]
You could possibly create temporary npm package at like `@vividlemon/bootstrap-vue3` until you regain access to the token?

**Comment by HenryUxDesugner** [comment]
Thank you @VividLemon . Also @AltF02 is it possible for you to generate a version of it in your own registry?

**Comment by AltF02** [comment]
@HenryUxDesugner possibly, I was running into some issues today, and unfortunately didn't have the time to figure them out. I will look further into it tomorrow :)

**Comment by VividLemon** [comment]
https://www.npmjs.com/package/bootstrap-vue-3
v0.4.6 has been released

The error with `ERR_PNPM_GIT_UNKNOWN_BRANCH` was the result of `npm-publish.yaml`, since it runs based on a **created release** it is not attached to a branch. It is attached to the HEAD. And therefore, cannot publish from HEAD

The error with `ERR_PNPM_GIT_NOT_UNCLEAN` (Now, from a report I made, was renamed to `ERR_PNPM_GIT_UNCLEAN`)... Was the result of the publishing suite running **tests** before publishing. This is normally not an issue. However, sometimes, when an eslint rule is not caught by a human beforehand, and it is fixed during this **publish** script, eslint will fix the issue and give it to the current running job. This causes an UNCLEAN branch, since the file and rule is "fixed" during this publish, inside the test... This has been fixed by removing the **testing** from the action. The ci.yaml action runs before and after each merge of a pr, so any testing should be caught there. It is not publish.yaml's responsibility

The error `npm ERR! code ENEEDAUTH` is the result of me removing `registry-url: 'https://registry.npmjs.org'` from the `actions/setup-node@v3`. When I originally went into to try to fix the above issues, I had updated each of the sub-actions. `actions/setup-node@v2` was updated to `@v3`. During this, I removed the necessary registry-url, causing the error.

This long list of issues has been fruitful however. The new ci system uses a caching system for node_modules. Making all of the action suites a bit faster. The publishing scripts should no longer have any errors with the removal of **testing** in them. In addition, the authors first suggestion of freezing the lockfile during actions has also been a great enhancement. 

The issue is now closed, the ci can now be considered fixed, and versions can hopefully be released more often.
